autoload/intero/process.vim

```diff
@@ -66,7 +66,13 @@ function! intero#process#initialize() abort
         if (!exists('g:intero_stack_yaml'))
             " Change dir temporarily and see if stack can find a config
             silent! lcd %:p:h
```

Comment:
From https://github.com/parsonsmatt/intero-neovim/issues/38#issuecomment-315565948

> One gotcha is that the STACK_YAML environment variable seems to be interpreted with respect to the file that is currently open.
> ... 
> So the value in `STACK_YAML` must be an absolute location, otherwise it'll be interpreted erroneously.

That's because we do this `lcd %:p:h` call in our code, effectively changing to the location where the Haskell file is, potentially invalidating the value of the `STACK_YAML` file.

In fact, I think we should leave the burden of finding the `stack.yaml` entirely up to `stack path`: we shouldn't `cd` ourselves at all. We should assume that if `STACK_YAML` is a relative path, it is valid with respect to **where NeoVim was launched**.

This will have the effect that the result that the user sees for `stack path --config-location` at the shell will be the same value that we see inside our plugin.